modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportStartReindexTaskAction.java

```diff
@@ -108,7 +108,7 @@ protected void doExecute(Task task, StartReindexTaskAction.Request request, Acti
 
         // In the current implementation, we only need to store task results if we do not wait for completion
         boolean storeTaskResult = request.getWaitForCompletion() == false;
-        ReindexTaskParams job = new ReindexTaskParams(storeTaskResult, included);
+        ReindexTaskParams job = new ReindexTaskParams(storeTaskResult, request.isResilient(), included);
```

Comment:
For a non-resilient reindex, I think we could change the failure message in [`waitForReindexDone` further down](https://github.com/elastic/elasticsearch/pull/51508/files#diff-e8832f8c1f07f4c537685cbc35535a7eR150) to signal more precisely that the original coordinator has failed?

I think this is the only real way to get assignment_failed out anyway, since we  now retry indefinitely.

Reply:
I added a new failure message in the case where assignment fails and resilient=false. I left a failure message when resilient=true. I don't think that scenario should happen. But it is nice to have the feedback in case in does happen and we can investigate.